### PR TITLE
feat(ProgressBar-ProgressRing): lightweight styling for ProgressBar-Ring

### PR DESCRIPTION
**Pull request by rajamatt**
﻿GitHub Issue: closes https://github.com/unoplatform/uno.themes-private/issues/6

## PR Type

What kind of change does this PR introduce?

- Feature

## Description

- Continues lightweight styling for ProgressBar and ProgressRing controls.

## PR Checklist 
Please check if your PR fulfills the following requirements:

- Commits must be following the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Tested UWP
- [x] Tested iOS
- [x] Tested Android
- [x] Tested WASM
- [ ] Tested MacOS
- [x] Contains **No** breaking changes
  > If the pull request contains breaking changes, commit message must contain a detailed description of the action to take for the consumer of this library. As explained by the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Media;
+using Uno.Extensions.Markup.Internals;
+
+namespace Uno.Themes.Markup
+{
+	public static partial class Theme
+	{
+		public static class ProgressBar
```
Comment (agneszitte) [inline src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs:9]: Missing **MinWidth** and **Height** to match XAML latest changes
https://github.com/unoplatform/Uno.Themes/blob/8c1f8b5c7958430b7fd785b4d9857ea4934a3270/src/library/Uno.Material/Styles/Controls/v2/ProgressBar.xaml#L12

Comment (rajamatt) [reply src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs:9]: @agneszitte where should this go in the class structure? Theme.ProgressBar.Resources.Default.**Size** ? And then have two properties MinWidth and Height

File: src/library/Uno.Themes.WinUI.Markup/Theme/ProgressRing.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace Uno.Themes.Markup
+{
+	public static partial class Theme
+	{
+		public static class ProgressRing
+		{
+			public static class Resources
+			{
+				public static class Default
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Theme/ProgressRing.cs:13]: We should not have a `Default` class here

Right @agneszitte ?

File: src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		{
+			public static class Resources
+			{
+				public static class Default
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs:13]: We should not have a Default class here

Right @agneszitte ?

File: src/library/Uno.Themes.WinUI.Markup/Theme/ProgressRing.cs (hunk 2 above)
Comment (Arieldelossantos) [reply src/library/Uno.Themes.WinUI.Markup/Theme/ProgressRing.cs]: removed

File: src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs (hunk 1 above)
Comment (Arieldelossantos) [reply src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs:9]: included

**Review by kazo0 — APPROVED** [review]

File: src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+				{
+					[ResourceKeyDefinition(typeof(Brush), "ProgressBarForeground")]
+					public static ResourceValue<Brush> Default => new("ProgressBarForeground", true);
+				}
+
+				public static class Background
+				{
+					[ResourceKeyDefinition(typeof(Brush), "ProgressBarBackground")]
+					public static ResourceValue<Brush> Default => new("ProgressBarBackground", true);
+				}
```
[the remark this answers is not in the document]
Comment (Marc-Antoine-Soucy) [reply src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs:23]: This makes sense for me.

File: src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+namespace Uno.Themes.Markup
+{
+	public static partial class Theme
+	{
+		public static class ProgressBar
+		{
+			public static class Resources
+			{
+				[ResourceKeyDefinition(typeof(Brush), "ProgressBarForeground")]
+				public static ResourceValue<Brush> Foreground => new("ProgressBarForeground", true);
```
Comment (kazo0) [inline src/library/Uno.Themes.WinUI.Markup/Theme/ProgressBar.cs:14]: Just needs to be updated with latest master and then replace ResourceValue with `ThemeResourceKey`

**Review by kazo0 — APPROVED** [review]

**Pull request merged**